src/metabase/pulse/render.clj

```diff
@@ -398,37 +398,40 @@
 
 (defn render-pulse-card
   "Render a single CARD for a `Pulse`. DATA is the `:data` from QP results (I think)."
-  [card data]
+  [card result]
   (try
-    [:a {:href   (card-href card)
-         :target "_blank"
-         :style  (style section-style
-                        {:margin          :16px
-                         :margin-bottom   :16px
-                         :display         :block
-                         :text-decoration :none})}
-     (when *include-title*
-       [:table {:style (style {:margin-bottom :8px
-                               :width         :100%})}
-        [:tbody
-         [:tr
-          [:td [:span {:style header-style}
-                (-> card :name h)]]
-          [:td {:style (style {:text-align :right})}
-           (when *include-buttons*
-             [:img {:style (style {:width :16px})
-                    :width 16
-                    :src   (render-image-with-filename "frontend_client/app/img/external_link.png")}])]]]])
-     (case (detect-pulse-card-type card data)
-       :empty     (render:empty     card data)
-       :scalar    (render:scalar    card data)
-       :sparkline (render:sparkline card data)
-       :bar       (render:bar       card data)
-       :table     (render:table     card data)
-       [:div {:style (style font-style
-                            {:color       "#F9D45C"
-                             :font-weight 700})}
-        "We were unable to display this card." [:br] "Please view this card in Metabase."])]
+    (when (:error result)
+      (throw (Exception. "Card has errors")))
```

Comment:
It might make sense to actually include the error message as part of the exception

```clojure
(when (:error result)
  (throw (Exception. (str "Card has errors: " (:error result)))))
```